src/smt/command.h

```diff
@@ -804,6 +808,27 @@ class CVC4_PUBLIC GetOptionCommand : public Command {
   std::string getCommandName() const throw();
 };/* class GetOptionCommand */
 
+// Set expression name command
+// Note this is not an official smt2 command
+// Conceptually:
+//   (assert (! expr :named name))
+// is converted to
+//   (assert expr)
+//   (set-expr-name expr name)
+class CVC4_PUBLIC SetExpressionNameCommand : public Command {
```

Comment:
Why not extend `AssertCommand` instead of adding `SetExpressionNameCommand`? In any case, we should probably modify the SMT2 printer to print out naming info, either by recognizing the `SetExpressionNameCommand` or by using the naming info from `SmtEngine` in the [`AssertCommand` printer](https://github.com/CVC4/CVC4/blob/master/src/printer/smt2/smt2_printer.cpp#L1287-L1289).

Reply:
The reason was that we may be interested in set expression names on terms.  For example, this is legal SMT LIB 2:

(assert (= (! (+ x y) :named a) 0))

...we ignore the name in this case, although perhaps later we may need to track these names for some reason?  I'm not sure.

Reply:
That makes sense, so we should change the printer. Thinking about it some more, we can also change the printer in another pull request if you prefer (since the current code doesn't do that either). If we decide to do that, we should open an issue for it.